util/rate_limiter_test.cc

```diff
@@ -390,21 +390,29 @@ TEST_F(RateLimiterTest, LimitChangeTest) {
           std::make_shared<GenericRateLimiter>(
               target, refill_period, 10, RateLimiter::Mode::kWritesOnly,
               SystemClock::Default(), false /* auto_tuned */);
-      ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->LoadDependency(
-          {{"GenericRateLimiter::Request",
-            "RateLimiterTest::LimitChangeTest:changeLimitStart"},
-           {"RateLimiterTest::LimitChangeTest:changeLimitEnd",
-            "GenericRateLimiter::RefillBytesAndGrantRequests"}});
-      Arg arg(target, Env::IO_HIGH, limiter);
+
       // The idea behind is to start a request first, then before it refills,
-      // update limit to a different value (2X/0.5X). No starvation should
-      // be guaranteed under any situation
+      // update limit to a different value (2X/0.5X). No starvation should be
+      // guaranteed under any situation
+      int32_t new_limit = (target << 1) >> (iter << 1);
+      ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->SetCallBack(
```

Comment:
I'm not really satisfied with the robustness of this test, because it looks like the test would keep passing if this callback were removed (e.g. through careless changes). How can we be sure this is testing what we think it's testing?

Also, is there a more realistic version of this test that does not require injecting unlock+lock?

Reply:
Reverted most of it.

> this test that does not require injecting unlock+lock

That injection is removed. Unlock+lock was kind of appealing because the mutex is released only during condition variable wait, and reacquired at the end. Since that's outside our code, I can't use regular sync points.